cadquery/occ_impl/exporters/threemf.py

```diff
@@ -0,0 +1,164 @@
+from datetime import datetime
+from os import PathLike
+import xml.etree.cElementTree as ET
+from typing import IO, List, Literal, Tuple, Union
+from zipfile import ZipFile, ZIP_DEFLATED, ZIP_STORED
+
+from ...cq import Compound, Shape, Vector
+
+
+class CONTENT_TYPES(object):
+    MODEL = "application/vnd.ms-package.3dmanufacturing-3dmodel+xml"
+    RELATION = "application/vnd.openxmlformats-package.relationships+xml"
+
+
+class SCHEMAS(object):
+    CONTENT_TYPES = "http://schemas.openxmlformats.org/package/2006/content-types"
+    RELATION = "http://schemas.openxmlformats.org/package/2006/relationships"
+    CORE = "http://schemas.microsoft.com/3dmanufacturing/core/2015/02"
+    MODEL = "http://schemas.microsoft.com/3dmanufacturing/2013/01/3dmodel"
+
+
+Unit = Literal["micron", "millimeter", "centimeter", "meter", "inch", "foot"]
+
+
+class ThreeMFWriter(object):
+    def __init__(
+        self,
+        shape: Shape,
+        tolerance: float,
+        angularTolerance: float,
+        unit: Unit = "millimeter",
+    ):
+        """
+        Initialize the writer.
+        Used to write the given Shape to a 3MF file.
+        """
+        self.unit = unit
+
+        if isinstance(shape, Compound):
+            shapes = list(shape)
+        else:
+            shapes = [shape]
+
+        tessellations = [s.tessellate(tolerance, angularTolerance) for s in shapes]
+        # Remove shapes that did not tesselate
+        self.tessellations = [t for t in tessellations if all(t)]
+
+    def write3mf(
+        self, outfile: Union[PathLike, str, IO[bytes]],
+    ):
+        """ 
+        Write to the given file. 
+        """
+
+        try:
```

Comment:
Is this really needed?

Reply:
Zipfile does not guarantee that zlib is present, and will throw an exception. From a quick Google search, I can see that this is a problem for some. It seems like this happens more commonly for people building python using homebrew on macs.